src/Google.Pubsub.V1/PublisherClient.cs

```diff
@@ -56,12 +57,80 @@ public sealed partial class PublisherSettings : ServiceSettingsBase
         /// <returns>A new instance of the default PublisherSettings.</returns>
         public static PublisherSettings GetDefault() => new PublisherSettings();
 
+        public static Predicate<RpcException> IdempotentRetryFilter { get; } =
+            RetrySettings.FilterForStatusCodes(StatusCode.Unavailable, StatusCode.DeadlineExceeded);
+
+        public static Predicate<RpcException> NonIdempotentRetryFilter { get; } =
+            RetrySettings.FilterForStatusCodes();
```

Comment:
The naming is unfortunate, but this isn't meant to the the negation on IdempotentRetryFilter.
I've taken the names directly from the config yaml.

IdempotentRetryFilter is which status codes to retry on for an idempotent RPC.
NonIdempotentRetryFilter is which status codes to retry on for a non-idempotent RPC. This is empty, so retry is not attempted on any response status codes.